crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,332 @@
+use glam::FloatExt;
+
+use crate::prelude::{Direction2d, Mat2, Vec2};
+
+/// A counterclockwise 2D rotation in radians.
+///
+/// The rotation angle is wrapped to be within the `]-pi, pi]` range.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+pub struct Rotation2d {
+    /// The cosine of the rotation angle in radians.
+    ///
+    /// This is the real part of the unit complex number representing the rotation.
+    pub cos: f32,
+    /// The sine of the rotation angle in radians.
+    ///
+    /// This is the imaginary part of the unit complex number representing the rotation.
+    pub sin: f32,
+}
+
+impl Default for Rotation2d {
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+impl Rotation2d {
+    /// No rotation.
+    pub const IDENTITY: Self = Self { cos: 1.0, sin: 0.0 };
+
+    /// A rotation of π radians.
+    pub const PI: Self = Self {
+        cos: -1.0,
+        sin: 0.0,
+    };
+
+    /// A counterclockwise rotation of π/2 radians.
+    pub const FRAC_PI_2: Self = Self { cos: 0.0, sin: 1.0 };
+
+    /// A counterclockwise rotation of π/3 radians.
+    pub const FRAC_PI_3: Self = Self {
+        cos: 0.5,
+        sin: 0.866_025_4,
```

Comment:
This value should be a constante to avoid "[magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))" in code.
Especially since this value appears multiple times.

Reply:
It's already in a constant in all these places, and it's pretty clear what its sin pi / ). This isn't what I think of as a magic number.

Reply:
The exact value is `sqrt(3) / 2`, but Rust doesn't have `SQRT_3` as a constant. I could add it or at least add a comment here though.

Reply:
I think I'll just leave it as is. Constants aren't ideal for trig function values since there'd have to be several constants with the same value because of symmetricity (e.g. `cos(pi/8) == sin(pi/3)`)